server/mdm/apple/apple_mdm.go

```diff
@@ -212,23 +214,34 @@ func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team
 	jsonProf.AwaitDeviceConfigured = true
 
 	depClient := NewDEPClient(d.depStorage, d.ds, d.logger)
-	res, err := depClient.DefineProfile(ctx, DEPName, &jsonProf)
+	// Get all relevant org names
+	var tmID *uint
+	if team != nil {
+		tmID = &team.ID
+	}
+	orgNames, err := d.ds.GetABMTokenOrgNamesForTeam(ctx, tmID)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "apple POST /profile request failed")
+		return ctxerr.Wrap(ctx, err, "getting org names for team to register profile")
 	}
 
-	if setupAsst != nil {
-		setupAsst.ProfileUUID = res.ProfileUUID
-		if err := d.ds.SetMDMAppleSetupAssistantProfileUUID(ctx, setupAsst.TeamID, res.ProfileUUID); err != nil {
-			return ctxerr.Wrap(ctx, err, "save setup assistant profile UUID")
-		}
-	} else {
-		var tmID *uint
-		if team != nil {
-			tmID = &team.ID
+	if len(orgNames) == 0 {
+		orgNames = append(orgNames, DEPName)
```

Comment:
That's to make current tests happy? I don't think we'll need that once everything is in the "new mode", right?

Reply:
Fixed this down here: https://github.com/fleetdm/fleet/pull/21517#discussion_r1732766622